### PR TITLE
CB-9049 Add list filtering with authorization

### PR DESCRIPTION
**Pull request by foldik**
Everyone starts reviewing can get a good understanding of the implementation by reading the authorization-common module's README.md file's list filtering section.

Filtered items:

environments,
datalakes,
datahubs,
freeipas,
credentials,
image catalogs,
recipes.

Cluster Definitions and Cluster Templates are not filtered at the moment.

Has to be merged after https://github.com/hortonworks/hortonworks-cloud/pull/2626.

### CODE REVIEW

**Comment by foldik** [comment]
More testing and real ums test configuration is needed. I'll add that later.

**Comment by foldik** [comment]
Filtered items:
- environments,
- datalakes,
- datahubs,
- freeipas,
- credentials,
- image catalogs,
- recipes.

**Review by horadla23 — COMMENTED** [review]
i tried to review everything, but there are lot of changes, maybe it is worth to held a review meeting for this PR, generally change looks good

File: auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/EntitlementService.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ public boolean isFmsToFreeipaBatchCallEnabled(String accountId) {
     }
 
     public boolean listFilteringEnabled(String accountId) {
-        return isEntitlementRegistered(accountId, CDP_LIST_FILTERING);
+        return isEntitlementRegistered(accountId, PERSONAL_VIEW_CB_BY_RIGHT);
```
[the remark this answers is not in the document]
Comment (foldik) [reply auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/EntitlementService.java:69]: We already have this entitlement in thunderhead.

File: auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/GrpcUmsClient.java [hunk 2]
```diff
@@ -489,12 +490,26 @@ public String rightCheckToString(RightCheck rightCheck) {
...
+    public List<Boolean> hasRightsOnResources(String actorCrn, String memberCrn, List<String> resourceCrns, String right, Optional<String> requestId) {
+        if (CollectionUtils.isEmpty(resourceCrns)) {
+            return List.of();
+        }
+        if (InternalCrnBuilder.isInternalCrn(memberCrn)) {
+            return resourceCrns.stream().map(r -> true).collect(Collectors.toList());
+        }
+        LOGGER.debug("Check if {} has rights on resources {}", actorCrn, resourceCrns);
+        PersonalResourceViewClient client = new PersonalResourceViewClient(channelWrapper.getChannel(), actorCrn, tracer);
+        List<Boolean> retVal = client.hasRightOnResources(RequestIdUtil.getOrGenerate(requestId), memberCrn, right, resourceCrns);
```
[the remark this answers is not in the document]
Comment (foldik) [reply auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/GrpcUmsClient.java:508]: yes

File: core/src/main/java/com/sequenceiq/cloudbreak/authorization/RecipeFiltering.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+public class RecipeFiltering extends AbstractAuthorizationFiltering<Set<RecipeView>> {
+
+    public static final String WORKSPACE_ID = "WORKSPACE_ID";
+
+    @Inject
+    private RecipeService recipeService;
+
+    public Set<RecipeView> filterRecipes(AuthorizationResourceAction action, Long workspaceId) {
+        Map<String, Object> args = new HashMap<>();
+        args.put(WORKSPACE_ID, workspaceId);
```
Comment (horadla23) [inline core/src/main/java/com/sequenceiq/cloudbreak/authorization/RecipeFiltering.java:31]: as discussed, if possible, we can extract workspace from threradlocal, and in that case we do not need to pass workspaceid from controller

File: core/src/main/java/com/sequenceiq/cloudbreak/service/image/ImageCatalogService.java [hunk 4]
```diff
@@ -127,6 +129,19 @@
         return imageCatalogs;
     }
 
+    public List<AuthorizationResource> findAsAuthorizationResorcesInWorkspace(Long workspaceId) {
+        return imageCatalogRepository.findAsAuthorizationResourcesInWorksapce(workspaceId);
```
Comment (horadla23) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/image/ImageCatalogService.java:133]: pls fix typo

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/context/TestContext.java [hunk 5]
```diff
@@ -329,6 +329,8 @@ T doAction(T entity, Class<? extends MicroserviceClient> clientClass, Action<T,
                 Log.then(null, assertion.getClass().getSimpleName() + " assertion is failed: " + ResponseUtil.getErrorMessage(e));
             }
             getExceptionMap().put(key, e);
+        } catch (Error e) {
```
Comment (aszegedi) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/context/TestContext.java:332]: Excellent catch! Thank you!

**Review by aszegedi — APPROVED** [review]

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/context/TestContext.java (hunk 5 above)
Comment (foldik) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/context/TestContext.java:332]: Thank you for the help!

**Review by horadla23 — APPROVED** [review]
LGTM

**Pull request merged**